config/src/config/network_config.rs

```diff
@@ -110,19 +110,11 @@ impl NetworkConfig {
         key.expect("identity key should be present")
     }
 
-    pub fn load(&mut self, network_role: RoleType) -> Result<(), Error> {
+    pub fn load(&mut self) -> Result<(), Error> {
         if self.listen_address.to_string().is_empty() {
             self.listen_address = utils::get_local_ip()
                 .ok_or_else(|| Error::InvariantViolation("No local IP".to_string()))?;
         }
-
-        if network_role.is_validator() {
-            crate::config::invariant(
-                self.network_peers.is_empty(),
-                "Validators should not define network_peers".into(),
-            )?;
-        }
-
```

Comment:
it wasn't really an invariant, just that it was previously pointless to set the trusted peers as a validator, since consensus would immediately clobber it with an `UpdateEligibleNodes`. with this diff, we can support multiple sources for trusted peers, so it's fine to add pubkeys from config.